core/controllers/admin.py

```diff
@@ -854,3 +854,37 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
+
+        if new_username is None or current_username is None:
+            raise self.InvalidInputException(
+                'Please ensure that the usernames are not empty.')
+
+        if type(current_username).__name__ != 'unicode' or type(
```

Comment:
Can we use `isinstance(current_username, str)`? [Also, are we expecting this to be a string?)

Reply:
We're actually expecting this to be unicode, but when I tried doing `isinstance(current_username, unicode)`, the following linter error got thrown:
`unicode built-in referenced (unicode-builtin)`, which is why I implemented the check this way.

Reply:
Can you please check how this is done in other places? [Have you checked python_utils.UNICODE?]

Reply:
[@aggarwalpulkit596, This thread might be helpful for you as well!]

Reply:
@DubeySandeep thanks

Reply:
> Can you please check how this is done in other places? [Have you checked python_utils.UNICODE?]

That worked, thanks @DubeySandeep!